source/customizing-application/editor-ui/creating-custom-editor.rst

```diff
@@ -1 +1,1165 @@
  onlinesrc directive.
          

- **forceLabel** (Optional)

  Force the label to be displayed for this action
  even if the action is not the first element of its
  kind. Label with always be displayed.

          

- **if** (Optional)

  An XPath expression to evaluate. If true, the control is displayed. eg.


  .. code-block:: xml

      count(gmd:MD_Metadata/gmd:identificationInfo/*/gmd:citation/gmd:CI_Citation/
      gmd:identifier[ends-with(gmd:MD_Identifier/gmd:code/gco:CharacterString,
      //gmd:MD_Metadata/gmd:fileIdentifier/gco:CharacterString)]) = 0


  will only displayed the action control if the resource identifier does not end
  with the metadata identifier.
        

- **class** (Optional)

  Optional CSS class to add to the parent div element. eg. gn-required to show a `*`.

- **btnLabel** (Optional)

  Optional label to be added to the button.

- **btnClass** (Optional)

  Optional CSS class to be added to the button.

- **or** (Optional)

  Local name to match if the element does not exist.

- **or** (Optional)

            
  The local name of the geonet child (ie. non existing element) to match.

  .. code-block:: xml

      <field xpath="/gmd:MD_Metadata/gmd:language"
             or="language"
             in="/gmd:MD_Metadata"/>



- **or** (Optional)



- **in** (Optional)

  XPath of the geonet:child element with the or name to look for. Usually
  points to the parent of last element of the XPath attribute.

- **in** (Optional)

  The element to search in for the geonet child.

- **addDirective** (Optional)

  The directive to use for the add control for this field.

Child elements:

- **template**, Optional element (see :ref:`creating-custom-editor-template`)

.. _creating-custom-editor-group:

Adding a group
--------------


A group of field


Attributes:

- **name** (Mandatory)
  
  Section identifier.
  Translations are set on client side.


A field on which user can do batch editing.


Attributes:

- **name** (Mandatory)

  
  Field identifier.
  Translations are set on client side.

- **xpath** (Mandatory)
                        
  XPath of the element to edit.

- **indexField** (Optional)

  Lucene index field name (as defined in dumpfields).
  The field will be used to preview current record values (TODO).

- **use** (Optional)

  Field type. Register here any Angular directive to be used
  on the client side. Default is simple text field.
                        
                      

- **removable** (Optional) Fixed value: **true**

                        
  Define if the field could be marked as deleted.

- **insertMode** (Optional)

  Define if the field should be insert or replace.
  Do not set this property for mandatory field (eg. title).

- **codelist** (Optional)

  The codelist identifier. eg. gmd:MD_TopicCategoryCode for topic category.

- **template**
 
  Define an XML template to use for the value to insert.
\ No newline at end of file
+.. _creating-custom-editor:

```

Comment:
This file is created from https://github.com/geonetwork/core-geonetwork/blob/main/schemas/config-editor.xsd using https://github.com/geonetwork/core-geonetwork/blob/main/docs/schema-doc/pom.xml#L63-L88 so it would be better to fix errors in the XSD source file.

Reply:
Will do, thanks! :smile: